blockingconcurrentqueue.h

```diff
@@ -362,6 +373,19 @@ namespace details
 				return tryWait() || waitWithPartialSpinning(timeout_usecs);
 			}
 
+      // Acquires either 0 or max, but never values in between
+      ssize_t tryWaitManyExclusive(ssize_t max) {
+				assert(max >= 0);
+				ssize_t oldCount = m_count.load(std::memory_order_relaxed);
+        while (true) {
```

Comment:
This loop can be simplified to:

    while (oldCount >= max) {
        if (m_count.compare_exchange_weak(oldCount, oldCount - max, std::memory_order_acquire, std::memory_order_relaxed))
            return max;
    }
    return 0;